ports/rp2/machine_rtc.c

```diff
@@ -94,7 +94,7 @@ STATIC mp_obj_t machine_rtc_datetime(mp_uint_t n_args, const mp_obj_t *args) {
             .year = mp_obj_get_int(items[0]),
             .month = mp_obj_get_int(items[1]),
             .day = mp_obj_get_int(items[2]),
-            .dotw = mp_obj_get_int(items[3]),
+            .dotw = timeutils_calc_weekday(t.year, t.month, t.day),
```

Comment:
is this legal C, to access members of a struct as it's being initialised?

Reply:
That is a good point. Instead of wondering, lets just do it outside of the main initialization.